modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkOperationModelParserSdk.java

```diff
@@ -0,0 +1,259 @@
+import org.mule.runtime.module.extension.internal.loader.parser.NestedRouteModelParser;
+import org.mule.runtime.module.extension.internal.loader.parser.OperationModelParser;
+import org.mule.runtime.module.extension.internal.loader.parser.OutputModelParser;
+import org.mule.runtime.module.extension.internal.loader.parser.ParameterGroupModelParser;
+import org.mule.runtime.module.extension.internal.loader.parser.StereotypeModelFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * {@link OperationModelParser} implementation for Mule SDK
+ *
+ * @since 4.5.0
+ */
+class MuleSdkOperationModelParserSdk extends BaseMuleSdkExtensionModelParser implements OperationModelParser {
+
+  private final ComponentAst operation;
+  private final TypeLoader typeLoader;
+
+  private String name;
+
+  public MuleSdkOperationModelParserSdk(ComponentAst operation, TypeLoader typeLoader) {
+    this.operation = operation;
+    this.typeLoader = typeLoader;
+
+    parseStructure();
+  }
+
+  private void parseStructure() {
+    name = getParameter(operation, "name");
+  }
+
+  private OutputModelParser asOutputModelParser(ComponentAst outputTypeElement) {
+    String type = getParameter(outputTypeElement, "type");
+
+    return typeLoader.load(type)
+        .map(mt -> new DefaultOutputModelParser(mt, false))
+        .orElseThrow(() -> new IllegalModelDefinitionException(format(
+                                                                      "Component <%s:%s> defines %s as '%s' but such type is not defined in the application",
+                                                                      outputTypeElement.getIdentifier().getNamespace(),
+                                                                      outputTypeElement.getIdentifier().getName(),
+                                                                      outputTypeElement.getIdentifier().getName())));
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public String getDescription() {
+    return this.<String>getOptionalParameter(operation, "description").orElse("");
+  }
+
+  @Override
+  public List<ModelProperty> getAdditionalModelProperties() {
+    return emptyList();
+  }
+
+  @Override
+  public List<String> getEmittedNotifications() {
+    // TODO: MULE-20075
+    return emptyList();
+  }
+
+  @Override
+  public OutputModelParser getOutputType() {
+    return asOutputModelParser(getOutputPayloadTypeElement());
+  }
+
+  @Override
+  public OutputModelParser getAttributesOutputType() {
+    return getOutputAttributesTypeElement()
+        .map(this::asOutputModelParser)
+        .orElse(VoidOutputModelParser.INSTANCE);
+  }
+
+  @Override
+  public List<ParameterGroupModelParser> getParameterGroupModelParsers() {
+    return getSingleChild(operation, "parameters")
+        .map(p -> getChildren(p, "parameter").collect(toList()))
```

Comment:
use `flatMap` instead of `map/collect`

Reply:
no, this isn't the typical flatMap use case. What I'm doing here is reducing the entire list of children (which first need to be converted to String) into a single `MuleSdkParameterGroupModelParser`

Reply:
as a matter of fact, the second map statement is from Optional, not Stream